File: 15 Dec 2025/code.cpp
class Solution {
  public:
    int cntWays(vector<int>& arr) {
        int n=arr.size();
        int so=0,se=0;
        for(int i=0;i<n;i++){if(i%2)so+=arr[i];else se+=arr[i];}
        int co=0,ce=0;
        int res=0;
        for(int i=0;i<n;i++){
            if(i%2)co+=arr[i],so-=arr[i];
            else ce+=arr[i],se-=arr[i];
            if(i%2){
                if(co-arr[i]+se == ce+so)res++;
            }   
            else{
                if(ce-arr[i]+so == co+se)res++;
            }
        }
        return res;
    }
};

File: 9 Jan 2026 Subarrays With At Most K Distinct Integers/code.py
class Solution:
    def countAtMostK(self, arr, k):
        j,n=0,len(arr)
        m={}
        res=0
        for i in range(n):
            m[arr[i]]=m.get(arr[i],0)+1
            while(len(m)>k):
                m[arr[j]]-=1
                if(m[arr[j]]==0):
                    del m[arr[j]]
                j+=1
            res+=(i-j+1)
        return res
        